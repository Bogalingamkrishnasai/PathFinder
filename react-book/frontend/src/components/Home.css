/* Home.css */
.home-content {
  background-color: #f8f9fa;
  min-height: 100vh;
  padding: 0 0px 40px;
}

.welcome-banner {
  background: linear-gradient(135deg, #2064b8, #7abcff);
  padding: 80px 20px;
  margin-bottom: 40px;
  text-align: center;
  color: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.welcome-message {
  font-size: 2.5rem;
  font-weight: 600;
  margin-bottom: 1rem;
  text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.1);
}

.welcome-subtitle {
  font-size: 1.8rem;
  font-weight: 400;
  margin-bottom: 2rem;
  opacity: 0.95;
}

.cta-button {
  display: inline-block;
  padding: 12px 30px;
  background-color: #ffd700;
  color: #024596;
  text-decoration: none;
  border-radius: 30px;
  font-weight: 600;
  transition: all 0.3s ease;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
}

.cta-button:hover {
  transform: translateY(-2px);
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
  background-color: #ffdf33;
}

.section-title {
  text-align: center;
  font-size: 2rem;
  color: #024596;
  margin: 40px 0;
  position: relative;
  padding-bottom: 10px;
}

.section-title::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  transform: translateX(-50%);
  width: 60px;
  height: 3px;
  background-color: #ffd700;
}

.home-component {
  max-width: 1200px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 30px;
  padding: 20px;
}

.appointment-section {
  max-width: 1200px;
  margin: 60px auto 0;
  padding: 40px 20px;
  background-color: white;
  border-radius: 12px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.05);
}

.appointment-links {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.appointment-actions {
  display: grid;
  gap: 20px;
  width: 100%;
  max-width: 600px;
}

.appointment-link {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 15px 25px;
  background-color: #024596;
  color: white;
  border-radius: 8px;
  text-decoration: none;
  transition: all 0.3s ease;
  font-weight: 500;
  gap: 10px;
}

.appointment-link:hover {
  background-color: #0066cc;
  transform: translateY(-2px);
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.1);
}

.appointment-link svg {
  width: 20px;
  height: 20px;
  fill: currentColor;
}

@media (min-width: 768px) {
  .welcome-message {
      font-size: 3rem;
  }
  
  .welcome-subtitle {
      font-size: 2.2rem;
  }
  
  .appointment-actions {
      grid-template-columns: repeat(2, 1fr);
  }
  
  .cta-button {
      padding: 15px 40px;
      font-size: 1.1rem;
  }

/* Roadmap Section Styles */
.roadmap-section {
  max-width: 1200px;
  margin: 60px auto 0;
  padding: 40px 20px;
  background-color: white;
  border-radius: 12px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.05);
  text-align: center;
}

.roadmap-btn {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 15px 30px;
  background-color: #024596;
  color: white;
  border: none;
  border-radius: 8px;
  font-size: 1rem;
  font-weight: 500;
  cursor: pointer;
  transition: all 0.3s ease;
  margin-top: 20px;
  gap: 10px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.roadmap-btn:hover {
  background-color: #0066cc;
  transform: translateY(-2px);
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.15);
}

.roadmap-btn:active {
  transform: translateY(0);
}

.roadmap-btn svg {
  width: 20px;
  height: 20px;
  fill: currentColor;
}


.job-sites-section {
  padding: 2rem;
  background-color: #f9f9f9;
  border-radius: 8px;
  margin: 2rem 0;
}

.job-sites-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  gap: 1.5rem;
  margin-top: 1.5rem;
}

.job-site-card {
  background: white;
  border-radius: 8px;
  padding: 1.5rem;
  text-align: center;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
  transition: transform 0.2s ease;
  color: #333;
  text-decoration: none;
}

.job-site-card:hover {
  transform: translateY(-5px);
  box-shadow: 0 4px 8px rgba(0,0,0,0.1);
}

.job-site-logo {
  width: 60px;
  height: 60px;
  margin: 0 auto 1rem;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
  font-weight: bold;
  font-size: 1.2rem;
}

.job-site-logo.linkedin { background-color: #0077b5; }
.job-site-logo.indeed { background-color: #2164f4; }
.job-site-logo.glassdoor { background-color: #0daa41; }
.job-site-logo.monster { background-color: #6d1f7a; }

.job-site-card h3 {
  margin: 0.5rem 0;
  color: #222;
}

.job-site-card p {
  margin: 0;
  color: #666;
  font-size: 0.9rem;
}

}

@media (min-width: 1024px) {
  .home-component {
      gap: 40px;
      padding: 40px;
  }
  .roadmap-btn {
    padding: 15px 40px;
    font-size: 1.1rem;
  }
}