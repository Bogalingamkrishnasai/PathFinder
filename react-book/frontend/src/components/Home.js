import { Link, useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import Thumbnail from './Thumbnail';
import { itemImages } from '../items';
import './Home.css';
import ItemType from '../types/item';

function Home({ items }) {
  const navigate = useNavigate();

  return (
    <div className="home-content">
      <div className="welcome-banner">
        <h1 className="welcome-message">Welcome to Pathfinder</h1>
        <h2 className="welcome-subtitle">Empowering Your Professional Journey</h2>
        <Link to="/Quiz" className="cta-button">
          Start Career Assessment â†’
        </Link>
      </div>

      <div className="courses-section">
        <h2 className="section-title">Featured Courses</h2>
        <div className="home-component">
          {items.map((item) => (
            <Thumbnail
              key={item.itemId}
              itemId={item.itemId}
              image={itemImages[item.imageId]}
              title={item.title}
              shortDescription={item.shortDescription}
            />
          ))}
        </div>
      </div>

      <div className="appointment-section">
        <h2 className="section-title">Mentorship Sessions</h2>
        <div className="appointment-links">
          <div className="appointment-actions">
            <a href="/schedule-appointment" className="appointment-link">
              <svg aria-hidden="true" viewBox="0 0 24 24">
                <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z" />
              </svg>
              Schedule Appointment
            </a>
            <a href="/manage-appointment" className="appointment-link">
              <svg className="icon" viewBox="0 0 24 24">
                <path d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM9 10H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm-8 4H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2z" />
              </svg>
              Manage Appointments
            </a>
          </div>
        </div>
      </div>

      <div className="roadmap-section">
        <h2 className="section-title">Career Path Exploration</h2>
        <button
          type="button"
          className="roadmap-btn"
          onClick={() => navigate('/Roadmap')}
        >
          <svg aria-hidden="true" viewBox="0 0 24 24">
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z" />
          </svg>
          View Career Roadmaps
        </button>
      </div>

      <div className="job-sites-section">
        <h2 className="section-title">Job Search Resources</h2>
        <div className="job-sites-grid">
          <a href="https://www.linkedin.com/jobs" target="_blank" rel="noopener noreferrer" className="job-site-card">
            <div className="job-site-logo">
              <img src="https://content.linkedin.com/content/dam/me/business/en-us/amp/brand-site/v2/bg/LI-Logo.svg.original.svg" alt="LinkedIn" width="120" />
            </div>
            <p>Professional networking and job opportunities</p>
          </a>
          <a href="https://www.indeed.com" target="_blank" rel="noopener noreferrer" className="job-site-card">
            <div className="job-site-logo">
              <img src="https://1000logos.net/wp-content/uploads/2023/01/Indeed-logo.png" alt="Indeed" width="120" />
            </div>
            <p>Millions of jobs from all over the web</p>
          </a>
          <a href="https://www.glassdoor.com" target="_blank" rel="noopener noreferrer" className="job-site-card">
            <div className="job-site-logo">
              <img src="https://1000logos.net/wp-content/uploads/2023/07/glassdoor-logo.webp" alt="Glassdoor" width="120" />
            </div>
            <p>Jobs with company reviews and salaries</p>
          </a>
          <a href="https://www.monster.com" target="_blank" rel="noopener noreferrer" className="job-site-card">
            <div className="job-site-logo">
              <img src="https://1000logos.net/wp-content/uploads/2024/04/Monster-Logo.png" alt="Monster" width="120" />
            </div>
            <p>Find jobs and career resources</p>
          </a>
          <a href="https://www.careerbuilder.com" target="_blank" rel="noopener noreferrer" className="job-site-card">
            <div className="job-site-logo">
              <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e6/CareerBuilder_Logo.svg/2560px-CareerBuilder_Logo.svg.png" alt="CareerBuilder" width="120" />
            </div>
            <p>Job search and recruitment platform</p>
          </a>
        </div>
      </div>
    </div>
  );
}

Home.propTypes = {
  items: PropTypes.arrayOf(ItemType).isRequired,
};

export default Home;
