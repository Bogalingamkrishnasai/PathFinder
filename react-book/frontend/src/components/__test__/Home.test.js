import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';
import '@testing-library/jest-dom';

// Mock the Thumbnail component
jest.mock('../Thumbnail', () => ({ itemId, image, title }) => (
  <div data-testid="thumbnail-mock">
    {title} - {itemId} - {image}
  </div>
));

// Mock the items module
jest.mock('../../items', () => ({
  itemImages: {
    img1: 'path/to/image1.jpg',
    img2: 'path/to/image2.jpg'
  }
}));

describe('Home Component', () => {
  const mockItems = [
    {
      itemId: '1',
      title: 'Course 1',
      description: 'Description 1',
      price: 99,
      salePrice: 79,
      imageId: 'img1'
    },
    {
      itemId: '2',
      title: 'Course 2',
      description: 'Description 2',
      price: 129,
      salePrice: 99,
      imageId: 'img2'
    }
  ];

  test('matches home component snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Home items={mockItems} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders welcome banner', () => {
    render(
      <BrowserRouter>
        <Home items={mockItems} />
      </BrowserRouter>
    );
    
    expect(screen.getByText('Welcome to Pathfinder')).toBeInTheDocument();
    expect(screen.getByText(/Start Career Assessment/)).toBeInTheDocument();
  });

  test('displays featured courses', () => {
    render(
      <BrowserRouter>
        <Home items={mockItems} />
      </BrowserRouter>
    );
    
    expect(screen.getByText('Featured Courses')).toBeInTheDocument();
    expect(screen.getAllByTestId('thumbnail-mock')).toHaveLength(2);
  });

  test('shows mentorship section', () => {
    render(
      <BrowserRouter>
        <Home items={mockItems} />
      </BrowserRouter>
    );
    
    expect(screen.getByText('Mentorship Sessions')).toBeInTheDocument();
    expect(screen.getByText('Schedule Appointment')).toBeInTheDocument();
  });

  test('displays job resources section', () => {
    render(
      <BrowserRouter>
        <Home items={mockItems} />
      </BrowserRouter>
    );
    
    expect(screen.getByText('Job Search Resources')).toBeInTheDocument();
    const linkedInElement = screen.getByRole('link', { name: /linkedin/i });
    expect(linkedInElement).toBeInTheDocument();
    expect(linkedInElement).toHaveAttribute('href', 'https://www.linkedin.com/jobs');
  });
});