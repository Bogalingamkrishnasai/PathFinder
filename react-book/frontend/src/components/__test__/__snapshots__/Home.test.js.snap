// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches home component snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-content"
  >
    <div
      class="welcome-banner"
    >
      <h1
        class="welcome-message"
      >
        Welcome to Pathfinder
      </h1>
      <h2
        class="welcome-subtitle"
      >
        Empowering Your Professional Journey
      </h2>
      <a
        class="cta-button"
        href="/Quiz"
      >
        Start Career Assessment â†’
      </a>
    </div>
    <div
      class="courses-section"
    >
      <h2
        class="section-title"
      >
        Featured Courses
      </h2>
      <div
        class="home-component"
      >
        <div
          data-testid="thumbnail-mock"
        >
          Course 1 - 1 - path/to/image1.jpg
        </div>
        <div
          data-testid="thumbnail-mock"
        >
          Course 2 - 2 - path/to/image2.jpg
        </div>
      </div>
    </div>
    <div
      class="appointment-section"
    >
      <h2
        class="section-title"
      >
        Mentorship Sessions
      </h2>
      <div
        class="appointment-links"
      >
        <div
          class="appointment-actions"
        >
          <a
            class="appointment-link"
            href="/schedule-appointment"
          >
            <svg
              aria-hidden="true"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
            Schedule Appointment
          </a>
          <a
            class="appointment-link"
            href="/manage-appointment"
          >
            <svg
              class="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM9 10H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm-8 4H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2z"
              />
            </svg>
            Manage Appointments
          </a>
        </div>
      </div>
    </div>
    <div
      class="roadmap-section"
    >
      <h2
        class="section-title"
      >
        Career Path Exploration
      </h2>
      <button
        class="roadmap-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
          />
        </svg>
        View Career Roadmaps
      </button>
    </div>
    <div
      class="job-sites-section"
    >
      <h2
        class="section-title"
      >
        Job Search Resources
      </h2>
      <div
        class="job-sites-grid"
      >
        <a
          class="job-site-card"
          href="https://www.linkedin.com/jobs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="job-site-logo"
          >
            <img
              alt="LinkedIn"
              src="https://content.linkedin.com/content/dam/me/business/en-us/amp/brand-site/v2/bg/LI-Logo.svg.original.svg"
              width="120"
            />
          </div>
          <p>
            Professional networking and job opportunities
          </p>
        </a>
        <a
          class="job-site-card"
          href="https://www.indeed.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="job-site-logo"
          >
            <img
              alt="Indeed"
              src="https://1000logos.net/wp-content/uploads/2023/01/Indeed-logo.png"
              width="120"
            />
          </div>
          <p>
            Millions of jobs from all over the web
          </p>
        </a>
        <a
          class="job-site-card"
          href="https://www.glassdoor.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="job-site-logo"
          >
            <img
              alt="Glassdoor"
              src="https://1000logos.net/wp-content/uploads/2023/07/glassdoor-logo.webp"
              width="120"
            />
          </div>
          <p>
            Jobs with company reviews and salaries
          </p>
        </a>
        <a
          class="job-site-card"
          href="https://www.monster.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="job-site-logo"
          >
            <img
              alt="Monster"
              src="https://1000logos.net/wp-content/uploads/2024/04/Monster-Logo.png"
              width="120"
            />
          </div>
          <p>
            Find jobs and career resources
          </p>
        </a>
        <a
          class="job-site-card"
          href="https://www.careerbuilder.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="job-site-logo"
          >
            <img
              alt="CareerBuilder"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e6/CareerBuilder_Logo.svg/2560px-CareerBuilder_Logo.svg.png"
              width="120"
            />
          </div>
          <p>
            Job search and recruitment platform
          </p>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
